Consolidation, correction, and minor revision of OW installation/setup documentation.

kmdalal opened the pull request:
Previously, various instructions (e.g. Plexil installation) were copied in 2-3 places, and with differences and errors.   Having most documentation in the wiki also affords direct editing; changing Markdown files required a branch, pull request, and approvals.

Samahu reviewed (changes requested):
I see some problems with this personally. The issue is that github wiki currently doesn't have a way to browse the docs of a give version. That being the case we will eventually run out of sync between the released version and the current state of wiki (which will certainly cause confusion to us and the users).

One approach to address this is by creating branches of the wiki on every release. Since github wiki doesn't yet support browsing different release. The main wiki would inform the users to  clone the wiki locally and checkout the relevant release docs

Samahu commented:
Alternatively we 

> I see some problems with this personally. The issue is that github wiki currently doesn't have a way to browse the docs of a give version. That being the case we will eventually run out of sync between the released version and the current state of wiki (which will certainly cause confusion to us and the users).
> 
> One approach to address this is by creating branches of the wiki on every release. Since github wiki doesn't yet support browsing different release. The main wiki would inform the users to clone the wiki locally and checkout the relevant release docs

Alternatively, the wiki may contain only docs that are tied to a specific release. Release specific docs would be part of the code.

kmdalal commented:
Good point.   I'll suggest this protocol:

1. The wiki always documents the master branch, but never anything that's not yet released.
2. Documentation for pre-release features goes in the repo, in Markdown files.
3. When a release is made, the wiki is updated with the new repo-located documentation, which is then removed.
4. When a release is made, a copy of the the wiki resulting from (3) is packaged with it (details TBD)

Samahu commented:
> Good point, Ussama. I'll suggest this protocol:
> 
> 1. The wiki always documents the master branch, but never anything that's not yet released.
> 2. Documentation for pre-release features goes in the repo, in Markdown files.
> 3. When a release is made, the wiki is updated with the new repo-located documentation, which is then removed.
> 4. When a release is made, a copy of the the wiki resulting from (3) is packaged with it (details TBD)

Seems reasonable to me

Samahu commented:
Another approach is to keep everything in wiki and create branches of the wiki repo per release.
when we create a new release we merge the branch into **master**

mogumbo commented:
I still prefer dropping this branch and doing the opposite: remove redundant setup info from the wiki and instead point to the readmes in the code. As Ussama points out, this is the easiest way to keep setup instructions tied to the version/branch they are associated with.

In general, I like to keep low-level documentation in the README.md files. This keeps documentation close to the code, making it more likely to stay up-to-date and making it easier for users to work offline. The wiki is a good place for additional info, such as high-level documentation, topics that relate to multiple packages or repos, and justifications/architecture decisions. Of course, there will always be some gray area that we need to deal with.

kmdalal commented:
Okay, I've moved the install/setup/build instructions back to the files they came from, and removed them from the wiki.   We can try moving forward with this scheme.

However, this branch and request is still valid: it consolidates information that was repeated, fixes errors, and has some slight rewording and reorganization.   Please give it a read ASAP.

mogumbo commented on oceanwaters/doc/setup_dev_env.md:
```diff
@@ -1,99 +1,54 @@
+    -   The Bash command shell
+    -   The git version control system
+-   The Robot Operating System (ROS), Melodic Morenia distribution, with:
+    -   Gazebo 9.13
```
Gazebo is not part of ROS and should be listed at the same bullet level.

kmdalal replied:
I think in all my many previous installations of the system, I got Gazebo (albeit the wrong version) when I was following instructions to install "ROS".  Nevertheless, I have made this edit.

mogumbo commented on oceanwaters/doc/setup_oceanwaters.md:
````diff
@@ -1,61 +1,97 @@
+
+NOTE: if catkin is not found, it was somehow left out of your ROS install.  Try:
 ```
 sudo apt-get install python-catkin-tools
````
python-catkin-tools is included in the extra packages installed in the environment setup, so I think we can remove this step.

kmdalal replied:
Done.  Myself and at least one other (I think Thomas) had to install catkin manually at this step, but that was probably because it was left out of the initial package list.

mogumbo commented on oceanwaters/doc/setup_oceanwaters.md:
````diff
@@ -1,61 +1,97 @@
-```
-source devel/setup.bash
-```
-* Launch the sim
````
I think launching the sim should be included. It assures the user that things compiled correctly.

kmdalal replied:
Done.  I felt some explanation of what to expect (which wasn't there before) was needed, so added something brief.

mogumbo commented on oceanwaters/doc/setup_oceanwaters.md:
```diff
@@ -1,61 +1,103 @@
+
+At this point you should see Gazebo related variables such as
+`GAZEBO_MODEL_PATH` set in your environment.  You may wish to add the line above to your
+.bashrc file.
```
Sorry, one more edit. I think sourcing setup.bash in your .bashrc is risky for most users. This would only be useful if you only use one ROS workspace and don't plan to recompile it much. I think this suggestion should be removed.

Samahu commented on oceanwaters/doc/setup_dev_env.md:
```diff
@@ -1,99 +1,54 @@
 
-Development Environment Setup
-=================
+The following minimum software set is required to build and run OceanWATERS
```
```suggestion
The following minimum software setup is required to build and run OceanWATERS
```

Samahu commented on oceanwaters/doc/setup_dev_env.md:
```diff
@@ -1,99 +1,54 @@
+    -   The git version control system
+-   The Robot Operating System (ROS), Melodic Morenia distribution, with:
+    -   The Catkin build system
+-   Gazebo 9.13
```
```suggestion
-   Gazebo 9.13 or higher
```

kmdalal commented:
I changed the name of this PR to reflect what it became.
